Gems/Atom/Tools/AtomToolsFramework/Code/Include/AtomToolsFramework/Viewport/ModularViewportCameraController.h

```diff
@@ -18,9 +18,22 @@ namespace AtomToolsFramework
 {
     class ModularViewportCameraControllerInstance;
 
+    //! A reduced ViewportContext interface for use by the ModularViewportCameraController.
+    //! @note This extra indirection is used to facilitate testing the ModularViewportCameraController.
+    class ModularCameraViewportContext
```

Comment:
I did go down the route of trying to instantiate a `ViewportContextManager` (but right now it has a private constructor and you need the whole `RPISystem` to be created...). It might be possible we can simplify it in future but with this we can limit what actually needs to be instantiated in the test which is nice. If there are more examples of setting up the RPI system for testing it would be interesting to see. Thanks!